diff --git a/node_modules/@astrojs/starlight/components/Head.astro b/node_modules/@astrojs/starlight/components/Head.astro
index ec2431f..487bf9c 100644
--- a/node_modules/@astrojs/starlight/components/Head.astro
+++ b/node_modules/@astrojs/starlight/components/Head.astro
@@ -13,6 +13,16 @@ const { data } = entry;
 
 const canonical = Astro.site ? new URL(Astro.url.pathname, Astro.site) : undefined;
 const description = data.description || config.description;
+const og = new URLSearchParams();
+
+const title = data.title || config.title;
+
+og.set("site", "PartyKit Docs");
+og.set("title", title);
+if (description && description !== "...") {
+  og.set("description", description);
+}
+const ogImage = `https://og.partykit.io/api/og?${og}`;
 
 const headDefaults: z.input<ReturnType<typeof HeadConfigSchema>> = [
 	{ tag: 'meta', attrs: { charset: 'utf-8' } },
@@ -50,6 +60,8 @@ const headDefaults: z.input<ReturnType<typeof HeadConfigSchema>> = [
 	},
 	{ tag: 'meta', attrs: { name: 'twitter:title', content: data.title } },
 	{ tag: 'meta', attrs: { name: 'twitter:description', content: description } },
+	{ tag: "meta", attrs: { name: "twitter:image", content: ogImage } },
+  { tag: "meta", attrs: { name: "og:image", content: ogImage } },
 ];
 
 if (description)
diff --git a/node_modules/@astrojs/starlight/integrations/expressive-code/theming.ts b/node_modules/@astrojs/starlight/integrations/expressive-code/theming.ts
index 39f96b1..a7491ec 100644
--- a/node_modules/@astrojs/starlight/integrations/expressive-code/theming.ts
+++ b/node_modules/@astrojs/starlight/integrations/expressive-code/theming.ts
@@ -1,5 +1,7 @@
 import { ExpressiveCodeTheme, type ThemeObjectOrShikiThemeName } from 'astro-expressive-code';
+// @ts-expect-error - missing types
 import nightOwlDark from './themes/night-owl-dark.jsonc?raw';
+// @ts-expect-error - missing types
 import nightOwlLight from './themes/night-owl-light.jsonc?raw';
 
 export type BundledThemeName = 'starlight-dark' | 'starlight-light';
