diff --git a/node_modules/@astrojs/starlight/components/HeadSEO.astro b/node_modules/@astrojs/starlight/components/HeadSEO.astro
index d187bd9..cb5d99f 100644
--- a/node_modules/@astrojs/starlight/components/HeadSEO.astro
+++ b/node_modules/@astrojs/starlight/components/HeadSEO.astro
@@ -18,6 +18,14 @@ const canonical = Astro.site ? new URL(Astro.url.pathname, Astro.site) : undefin
 const title = data.title || config.title;
 const description = data.description || config.description;
 
+const og = new URLSearchParams();
+og.set("site", "PartyKit Docs");
+og.set("title", title);
+if (description && description !== "...") {
+  og.set("description", description);
+}
+const ogImage = `https://og.partykit.io/api/og?${og}`;
+
 const headDefaults: z.input<ReturnType<typeof HeadConfigSchema>> = [
 	{ tag: 'meta', attrs: { charset: 'utf-8' } },
 	{
@@ -54,6 +62,8 @@ const headDefaults: z.input<ReturnType<typeof HeadConfigSchema>> = [
 	},
 	{ tag: 'meta', attrs: { name: 'twitter:title', content: title } },
 	{ tag: 'meta', attrs: { name: 'twitter:description', content: description } },
+	{ tag: "meta", attrs: { name: "twitter:image", content: ogImage } },
+  { tag: "meta", attrs: { name: "og:image", content: ogImage } },
 ];
 
 if (description)
