diff --git a/node_modules/@cloudflare/ai/dist/index.js b/node_modules/@cloudflare/ai/dist/index.js
index dd442fd..cea8706 100644
--- a/node_modules/@cloudflare/ai/dist/index.js
+++ b/node_modules/@cloudflare/ai/dist/index.js
@@ -2772,10 +2772,12 @@ var Ai = class {
               "content-type": "application/json"
             }
           };
-          const res = await fetch(
-            `https://api.cloudflare.com/client/v4/accounts/${this.options.apiAccount}/ai/run/${model}`,
-            fetchOptions
-          );
+          // const res = await fetch(
+          //   `https://api.cloudflare.com/client/v4/accounts/${this.options.apiAccount}/ai/run/${model}`,
+          //   fetchOptions
+          // );
+          const apiGateway = typeof this.options.apiGateway === "string" ? this.options.apiGateway : "https://ai.labs.partykit.dev";
+          const res = await fetch(`${apiGateway}/${model}`, fetchOptions);
           if (!res.ok) {
             throw new Error(await res.text());
           }
