diff --git a/packages/partykit-ai/node_modules/@cloudflare/ai/dist/index.js b/packages/partykit-ai/node_modules/@cloudflare/ai/dist/index.js
index 285bf28..26a9fad 100644
--- a/packages/partykit-ai/node_modules/@cloudflare/ai/dist/index.js
+++ b/packages/partykit-ai/node_modules/@cloudflare/ai/dist/index.js
@@ -3218,10 +3218,12 @@ var Ai = class {
               "content-type": "application/json"
             }
           };
-          const res = await fetch(
-            `https://api.cloudflare.com/client/v4/accounts/${this.options.apiAccount}/ai/run/${model}`,
-            fetchOptions
-          );
+          // const res = await fetch(
+          //   `https://api.cloudflare.com/client/v4/accounts/${this.options.apiAccount}/ai/run/${model}`,
+          //   fetchOptions
+          // );
+          const apiGateway = typeof this.options.apiGateway === "string" ? this.options.apiGateway : "https://ai.labs.partykit.dev";
+          const res = await fetch(`${apiGateway}/${model}`, fetchOptions);
           if (!res.ok) {
             throw new Error(await res.text());
           }
